core/extensions.rs

```diff
@@ -584,21 +567,21 @@ impl ExtensionBuilder {
 #[cfg(not(feature = "include_js_files_for_snapshotting"))]
 #[macro_export]
 macro_rules! include_js_files {
-  (dir $dir:literal, $($file:literal,)+) => {
+  ($name:ident dir $dir:literal, $($file:literal,)+) => {
```

Comment:
Sorry for that @bajtos. Could you help me spot here what's the mistake in docs?

Reply:
@bartlomieju sorry for posting a comment that's not actionable; I'll do better next time 🙈 

The new version of `include_js_files!` requires the extension name (`$name:ident`) as the first argument.

I think this is the correct example:

```rs
include_js_files!(
   "my-extension",
   dir "js",
   "01_hello.js",
   "02_goodbye.js",
)
```

In the end, I decided to bite the bullet and upgrade my code to use `deno_core::extension!` macro. It wasn't that much extra work, kudos for making the `extension!` macro easy to use outside of the Deno repo too!

Reply:
Ah I see, sorry again for the breaking change. We did all these changes to improve startup performance - both for Deno itself and all embedders. The `extension!` macro moves as much of the work to "build time" instead of "startup time" so hopefully you can see some reduction in your startup time too. Kudos go to @mmastrac for the macro :)

Reply:
The changes make sense; I understand why you are making them 👍🏻 

Speaking about startup performance - is there any guide for embedders? I am particularly interested in learning how to set up our build to create a snapshot of all JS files we are bundling in our runtime to get that faster startup time you mentioned. I can probably figure this out by reading the source code of Deno core & cli, but it would be much nicer if there were a guide. Having said that, both the guide and snapshotting are pretty much "nice to have" for us right now.

Reply:
Done: https://github.com/denoland/deno/issues/18503